File: server/src/main/java/com/dogwalk/server/domain/member/MemberController.java
package com.dogwalk.server.domain.member;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.List;


@RestController
@RequiredArgsConstructor
@RequestMapping("members")
public class MemberController {
    private final MemberService memberService;

    @PostMapping
    public ResponseEntity<Void> signup(@RequestBody @Valid final MemberRequest dto) {
        memberService.create(dto);
        return ResponseEntity.status(HttpStatus.CREATED).build();
    }

    @GetMapping(params = "id")
    public ResponseEntity<MemberResponse> getById(@RequestParam final Long id) {
        return ResponseEntity.ok(memberService.findById(id));
    }

    @GetMapping(params = "name")
    public ResponseEntity<List<MemberResponse>> getByName(@RequestParam final String name) {
        return ResponseEntity.ok(memberService.findByPetName(name));
    }
}

File: kotlin-server/src/main/kotlin/com/example/kotlinserver/member/MemberController.kt
package com.example.kotlinserver.member

import com.example.kotlinserver.member.dto.MemberResponse
import com.example.kotlinserver.member.dto.SignUpRequest
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/members")
class MemberController(
    private val memberService: MemberService
) {
    @PostMapping
    fun signUp(@RequestBody dto: SignUpRequest): ResponseEntity<Void> {
        memberService.save(dto)
        return ResponseEntity.status(HttpStatus.CREATED).build()
    }

    @GetMapping
    fun searchList(@RequestParam(value = "") nickname: String): ResponseEntity<List<MemberResponse>> {
        return ResponseEntity.ok(memberService.findAllByNicknameContaining(nickname));
    }
}